### PR TITLE
fix iMigration type hints

### PR DESCRIPTION
**Pull request by rullzer**

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]
Thanks Roeland, I just ran into this

**Comment by rullzer** [comment]
> Thanks Roeland, I just ran into this

well it still does boom it seems.
But feel free ot have a look and push potential fixes ;)

File: lib/public/Migration/SimpleMigrationStep.php [hunk 1]
```diff
@@ -53,25 +55,29 @@ public function description(): string {
...
-	 * {@inheritDoc}
-	 *
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
 	 * @since 13.0.0
 	 */
-	public function changeSchema(IOutput $output, \Closure $schemaClosure, array $options) {
+	public function changeSchema(IOutput $output, \Closure $schemaClosure, array $options): ?ISchemaWrapper {
```
Comment (miaulalala) [inline lib/public/Migration/SimpleMigrationStep.php:73]: I think this will make all old migrations go kaboom...

**Review by miaulalala — CHANGES_REQUESTED** [review]

File: core/Command/Db/Migrations/GenerateCommand.php [hunk 2]
```diff
@@ -92,16 +89,16 @@ public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $
 	 * @param array $options
 	 * @return null|ISchemaWrapper
 	 */
-	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
```
Comment (miaulalala) [inline core/Command/Db/Migrations/GenerateCommand.php:95]: this now makes static analysis go :fire: in migration generation.

**Comment by miaulalala** [comment]
@rullzer feel free to continue but I'm out, this breaks all apps unfortunately

**Pull request closed without merging**